wolfcrypt/src/wc_port.c

```diff
@@ -423,6 +484,37 @@ void wc_ReadDirClose(ReadDirCtx* ctx)
 
 #endif /* !NO_FILESYSTEM && !NO_WOLFSSL_DIR */
 
+#if !defined(NO_FILESYSTEM) && defined(WOLFSSL_ZEPHYR)
+XFILE z_fs_open(const char* filename, const char* perm)
+{
+    XFILE file;
+
+    file = XMALLOC(sizeof(file), NULL, DYNAMIC_TYPE_FILE);
```

Comment:
This still doesn't look right.  file is already sizeof(file), why would we need to dynamically allocate the same size as the object on the stack?

Has this code been tested?

Reply:
Fixed.